monitoring/system_pods.go

```diff
@@ -104,17 +105,21 @@ func (r *systemPodsChecker) check(ctx context.Context, reporter health.Reporter)
 
 // getPods returns a list of the local pods that have the
 // `gravitational.io/critical-pod` label.
-func (r *systemPodsChecker) getPods() ([]corev1.Pod, error) {
+func (r *systemPodsChecker) getPods() (pods []corev1.Pod, err error) {
 	opts := metav1.ListOptions{
 		LabelSelector: systemPodsSelector.String(),
 		FieldSelector: fields.OneTermEqualSelector("spec.nodeName", r.NodeName).String(),
 	}
-	pods, err := r.Client.CoreV1().Pods(kubernetes.AllNamespaces).List(opts)
-	if err != nil {
-		return nil, utils.ConvertError(err)
+
+	for _, namespace := range r.Namespaces {
```

Comment:
Is AllNamespaces still supposed to be used as a fallback for empty configuration (`len(r.Namespaces) == 0`)?

Reply:
I think it would be better to just skip the check if an empty configuration is provided. If we default to AllNamespaces we may risk users experiencing the issue that this PR was trying to solve.